lib/improver/cli/standardise.py

```diff
@@ -32,181 +32,41 @@
-
-    if (args.input_landmask_filepath and
-            "nearest-with-mask" not in args.regrid_mode):
-        msg = ("Land-mask file supplied without appropriate regrid_mode. "
-               "Use --regrid_mode=nearest-with-mask.")
-        raise ValueError(msg)
-
-    if args.input_landmask_filepath and not args.target_grid_filepath:
-        msg = ("Cannot specify input_landmask_filepath without "
-               "target_grid_filepath")
-        raise ValueError(msg)
-
-    # Load Cube and json
-    attributes_dict = load_json_or_none(args.json_file)
-    # source file data path is a mandatory argument
-    output_data = load_cube(args.source_data_filepath)
-    target_grid = None
-    source_landsea = None
-    if args.target_grid_filepath:
-        target_grid = load_cube(args.target_grid_filepath)
-        if args.regrid_mode in ["nearest-with-mask"]:
-            if not args.input_landmask_filepath:
-                msg = ("An argument has been specified that requires an input "
-                       "landmask filepath but none has been provided")
-                raise ValueError(msg)
-            source_landsea = load_cube(args.input_landmask_filepath)
-
-    # Process Cube
-    output_data = process(output_data, target_grid, args.regrid_mode,
-                          args.extrapolation_mode, source_landsea,
-                          args.landmask_vicinity, args.regridded_title,
-                          attributes_dict, args.coords_to_remove,
-                          args.new_name, args.new_units, args.fix_float64)
-
-    # Save Cube
-    if args.output_filepath:
-        save_netcdf(output_data, args.output_filepath)
-
-
-def process(output_data, target_grid=None, regrid_mode='bilinear',
-            extrapolation_mode='nanmask', source_landsea=None,
-            landmask_vicinity=25000, regridded_title=None,
-            attributes_dict=None, coords_to_remove=None, new_name=None,
-            new_units=None, fix_float64=False):
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(source_data: cli.inputcube,
+            *,
+            target_grid: cli.inputcube = None,
+            source_landmask: cli.inputcube = None,
+            regrid_mode='bilinear',
+            extrapolation_mode='nanmask',
+            landmask_vicinity=25000.,
+            regridded_title: str = None,
+            attributes_dict: cli.inputjson = None,
+            coords_to_remove: cli.comma_separated_list = None,
+            new_name: str = None,
+            new_units: str = None,
+            fix_float64=False):
     """Standardises a cube by one or more of regridding, updating meta-data etc
 
-    Standardise a source cube. Available options are regridding
-    (bi-linear or nearest-neighbour, optionally with land-mask
-    awareness), updating meta-data and converting float64 data to
-    float32. A check for float64 data compliance can be made by only
-    specifying a source cube with no other arguments.
```

Comment:
I think it would be a good idea to retain this last sentence as it is still true and is still a useful function.

Reply:
Well yes it's true, but it's not the whole truth, as that's not the only thing the standardise plugin does.  So I believe this is misleading, which is why I removed it.

Reply:
ok